src/tiled-layer.ts

```diff
@@ -78,8 +78,29 @@ export class TiledLayer extends TiledEntity {
     */
    public compression?: TiledCompression;
 
+   /**
+    * Original order of the Tiled layer
+    */
+   public order!: number;
+
    /**
     * Reference to the raw tiled layer data
     */
    public rawLayer!: RawTiledLayer;
+
+   public static parse(layer: RawTiledLayer): TiledLayer {
+      if (layer.type !== 'tilelayer') throw Error('Cannot parse a non tilelayer type layer');
+      const resultLayer = new TiledLayer();
+      resultLayer.id = +layer.id;
```

Comment:
I get this is a clever way to coerce but it's not clear what this does at first glance especially for someone not as familiar with hacks in JS.

Might as well use the standard, it's available in all browsers we support and it's clearer.


```suggestion
      resultLayer.id = Number.parseInt(layer.id);
```

Note radix is optional and is 10 by default anyway!

Note also that there was code before that didn't coerce, because this one character can get lost in the noise.